### ISSUE TITLE
Feature request: Connect to Remote

### ISSUE THREAD

**Issue by timbmg** [Dando18/slurm-dashboard#137]
It would be nice if I could connect from my local VS Code to a remote cluster where Slurm is running. I usually do not let the VS code run on the remote because it consumes quite some resources, blocking our login node if many people are doing so.

**Comment by Dando18** [comment]
I agree this would be nice. One of the clusters I work on a lot struggles to run the vscode server and it causes a lot of issues. However, I think this functionality would be difficult to implement and a bit out of scope for this extension.

Currently the extension interfaces with VSCode's APIs to see files and execute commands. The extension has to be running on the host where the files and commands live (sbatch, squeue, etc in this case). I guess you could theoretically wrap all the system calls in `ssh user@host "sbatch ..."`, but this would be error-prone and break from vscode's current nice virtualizations for remote workspaces.

It might be better to try this ssh trick outside the extension i.e. aliasing `sbatch` to `ssh user@host "sbatch ..."` on your local machine. This way when the extension tries to run slurm calls it will run on the remote host. You could even pass local files over ssh since sbatch can take files via stdin, i.e. `ssh user@host "sbatch" < local_file`. It's a little hacky, but this might be the best way to set this up.

Let me know what you think. I'm happy to try messing around with the above setup at some point as well.
